### PR TITLE
Added publish options for device and boundle nodes

### PR DESCRIPTION
**Pull request by mimi89999**

### CODE REVIEW

**Comment by chrisballinger** [comment]
Looks like there are some failing tests:

```
OMEMOElementTests

  testBundleParsing, [31m(([iq XMLStringWithOptions:NSXMLNodePrettyPrint]) equal to ([expectedXML XMLStringWithOptions:NSXMLNodePrettyPrint])) failed: ("<iq type="set" id="announce2">[0m
```

**Comment by mimi89999** [comment]
I updated the test. Could you please review this pull request?

File: Extensions/OMEMO/XMPPIQ+OMEMO.m [hunk 1]
```diff
@@ -71,6 +84,33 @@ + (XMPPIQ*) omemo_iqPublishDeviceIds:(NSArray<NSNumber*>*)deviceIds elementId:(n
...
+    [x addAttributeWithName:@"type" stringValue:@"submit"];
+    
+    NSXMLElement *formTypeField = [NSXMLElement elementWithName:@"field"];
+    [formTypeField addAttributeWithName:@"var" stringValue:@"FORM_TYPE"];
+    [formTypeField addAttributeWithName:@"type" stringValue:@"hidden"];
+    [formTypeField addChild:[NSXMLElement elementWithName:@"value" stringValue:XMLNS_PUBSUB_PUBLISH_OPTIONS]];
+    
+    [x addChild:formTypeField];
+    
+    NSXMLElement *PersistanceField = [NSXMLElement elementWithName:@"field"];
```
Comment (chrisballinger) [inline Extensions/OMEMO/XMPPIQ+OMEMO.m]: Would prefer lowercase `persistanceField`

File: Extensions/OMEMO/XMPPIQ+OMEMO.m [hunk 2]
```diff
@@ -71,6 +84,33 @@ + (XMPPIQ*) omemo_iqPublishDeviceIds:(NSArray<NSNumber*>*)deviceIds elementId:(n
...
+    [PersistanceField addAttributeWithName:@"var" stringValue:@"pubsub#persist_items"];
+    [PersistanceField addChild:[NSXMLElement elementWithName:@"value" objectValue:@"1"]];
+    
+    [x addChild:PersistanceField];
+    
+    NSXMLElement *AccessModelField = [NSXMLElement elementWithName:@"field"];
```
Comment (chrisballinger) [inline Extensions/OMEMO/XMPPIQ+OMEMO.m]: would prefer `accessModelField`

File: Extensions/OMEMO/XMPPIQ+OMEMO.m [hunk 3]
```diff
@@ -156,6 +209,33 @@ + (XMPPIQ*) omemo_iqPublishBundle:(OMEMOBundle*)bundle
...
+    [x addAttributeWithName:@"type" stringValue:@"submit"];
+    
+    NSXMLElement *formTypeField = [NSXMLElement elementWithName:@"field"];
+    [formTypeField addAttributeWithName:@"var" stringValue:@"FORM_TYPE"];
+    [formTypeField addAttributeWithName:@"type" stringValue:@"hidden"];
+    [formTypeField addChild:[NSXMLElement elementWithName:@"value" stringValue:XMLNS_PUBSUB_PUBLISH_OPTIONS]];
+    
+    [x addChild:formTypeField];
+    
+    NSXMLElement *PersistanceField = [NSXMLElement elementWithName:@"field"];
```
Comment (chrisballinger) [inline Extensions/OMEMO/XMPPIQ+OMEMO.m]: `persistanceField`

File: Extensions/OMEMO/XMPPIQ+OMEMO.m [hunk 4]
```diff
@@ -156,6 +209,33 @@ + (XMPPIQ*) omemo_iqPublishBundle:(OMEMOBundle*)bundle
...
+    [PersistanceField addAttributeWithName:@"var" stringValue:@"pubsub#persist_items"];
+    [PersistanceField addChild:[NSXMLElement elementWithName:@"value" objectValue:@"1"]];
+    
+    [x addChild:PersistanceField];
+    
+    NSXMLElement *AccessModelField = [NSXMLElement elementWithName:@"field"];
```
Comment (chrisballinger) [inline Extensions/OMEMO/XMPPIQ+OMEMO.m]: `accessModelField`

**Comment by chrisballinger** [comment]
Thanks for the bump, looks good! I haven't tested it on my end yet but I think it's good to merge. If you want to make changes regarding my comments above I'll hold off on merging until I hear back from you, but it's not a dealbreaker either way.

**Comment by mimi89999** [comment]
I changed the names to lowercase.

**Comment by mimi89999** [comment]
Thanks

**Pull request merged**